src/tools/perf/lib/ucp_tests.cc

```diff
@@ -496,35 +528,44 @@ class ucp_perf_test_runner {
     ucs_status_t send_last_iter(ucp_ep_h ep, void *buffer, size_t size,
                                 uint64_t remote_addr, ucp_rkey_h rkey)
     {
+        uint64_t atomic_param = 0;
         ucs_status_ptr_t status_p;
 
         if (!use_psn()) {
             return UCS_OK;
         }
 
         fence();
-        *(uint8_t*)buffer = UCP_PERF_LAST_ITER_SN;
+
+        /* Make sure that doing the last opetarion will write 1 to the end of
+           the remote buffer */
+        if (CMD == UCX_PERF_CMD_PUT) {
+            write_sn(buffer, m_perf.params.send_mem_type, size, LAST_ITER_SN);
+        } else if (is_atomic()) {
+            write_sn(&atomic_param, UCS_MEMORY_TYPE_HOST, size, LAST_ITER_SN);
+        }
+
         /* coverity[switch_selector_expr_is_constant] */
         switch (CMD) {
         case UCX_PERF_CMD_PUT:
-            return ucp_put(ep, buffer, sizeof(uint8_t), remote_addr, rkey);
+            return ucp_put(ep, buffer, size, remote_addr, rkey);
         case UCX_PERF_CMD_ADD:
-            return ucp_atomic_post(ep, UCP_ATOMIC_POST_OP_ADD, 1, size,
-                                   remote_addr, rkey);
+            return ucp_atomic_post(ep, UCP_ATOMIC_POST_OP_ADD, atomic_param,
+                                   size, remote_addr, rkey);
         case UCX_PERF_CMD_FADD:
-            status_p = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_FADD, 1,
-                                           buffer, size, remote_addr, rkey,
-                                           nop_cb);
+            status_p = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_FADD,
+                                           atomic_param, buffer, size,
+                                           remote_addr, rkey, free_cb);
             return UCS_PTR_STATUS(status_p);
         case UCX_PERF_CMD_SWAP:
-            status_p = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_SWAP, 1,
-                                           buffer, size, remote_addr, rkey,
-                                           nop_cb);
+            status_p = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_SWAP,
```

Comment:
maybe unite fadd & cswap, use m_atomic_op?

Reply:
in order to use m_atomic_op, used ucp_atomic_op_nbx() instead of ucp_atomic_fetch_nb()